src/instana/instrumentation/pika.py

```diff
@@ -2,167 +2,244 @@
 # (c) Copyright IBM Corp. 2021
 # (c) Copyright Instana Inc. 2021
 
+import types
+from typing import TYPE_CHECKING, Any, Callable, Dict, Generator, Optional, Tuple, Union
 
-
+import pika.connection
 import wrapt
-import opentracing
-import types
 
-from ..log import logger
-from ..singletons import tracer
-from ..util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.log import logger
+from instana.propagators.format import Format
+from instana.singletons import tracer
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
+
+if TYPE_CHECKING:
+    from instana.span.span import InstanaSpan
 
 try:
     import pika
 
-
-    def _extract_broker_tags(span, conn):
-        span.set_tag("address", "%s:%d" % (conn.params.host, conn.params.port))
-
-
-    def _extract_publisher_tags(span, conn, exchange, routing_key):
+    if TYPE_CHECKING:
+        import pika.adapters.blocking_connection
+        import pika.channel
+
+    def _extract_broker_tags(
+        span: "InstanaSpan", conn: pika.connection.Connection
+    ) -> None:
+        span.set_attribute("address", f"{conn.params.host}:{conn.params.port}")
+
+    def _extract_publisher_tags(
+        span: "InstanaSpan",
+        conn: pika.connection.Connection,
+        exchange: str,
+        routing_key: str,
+    ) -> None:
         _extract_broker_tags(span, conn)
 
-        span.set_tag("sort", "publish")
-        span.set_tag("key", routing_key)
-        span.set_tag("exchange", exchange)
-
+        span.set_attribute("sort", "publish")
+        span.set_attribute("key", routing_key)
+        span.set_attribute("exchange", exchange)
 
-    def _extract_consumer_tags(span, conn, queue):
+    def _extract_consumer_tags(
+        span: "InstanaSpan", conn: pika.connection.Connection, queue: str
+    ) -> None:
         _extract_broker_tags(span, conn)
 
-        span.set_tag("sort", "consume")
-        span.set_tag("queue", queue)
-
-
-    @wrapt.patch_function_wrapper('pika.channel', 'Channel.basic_publish')
-    def basic_publish_with_instana(wrapped, instance, args, kwargs):
-        def _bind_args(exchange, routing_key, body, properties=None, *args, **kwargs):
+        span.set_attribute("sort", "consume")
+        span.set_attribute("queue", queue)
+
+    @wrapt.patch_function_wrapper("pika.channel", "Channel.basic_publish")
+    def basic_publish_with_instana(
+        wrapped: Callable[..., pika.channel.Channel.basic_publish],
+        instance: pika.channel.Channel,
+        args: Tuple[object, ...],
+        kwargs: Dict[str, Any],
+    ) -> object:
+        def _bind_args(
+            exchange: str,
+            routing_key: str,
+            body: str,
+            properties: Optional[object] = None,
+            *args: Tuple[object, ...],
+            **kwargs: Dict[str, Any],
```

Comment:
Since `*args` and `**kwargs` represent the unpacked value inside the `Tuple`/ `Dict`, I guess `object` or a `Union` of known types might be a better value.

Slack discussion we had sometime back: [here](https://instana.slack.com/archives/C07DTNKMFDF/p1721809080802609)
```suggestion
            *args: object,
            **kwargs: object,
```

Reply:
Done